sktime/forecasting/base/adapters/_darts.py

```diff
@@ -0,0 +1,409 @@
+"""Implements adapter for Darts models."""
+
+import abc
+from typing import Optional
+
+import pandas as pd
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+
+__author__ = ["yarnabrina", "fnhirwa"]
+
+
+class _DartsRegressionModelsAdapter(BaseForecaster):
+    """Base adapter class for Darts Regression models.
+
+    Parameters
+    ----------
+    past_covariates : Optional[List[str]], optional
+        column names in ``X`` which are known only for historical data, by default None
+    num_samples : Optional[int], optional
+        Number of times a prediction is sampled from a probabilistic model, by default
+        1000
+
+    Notes
+    -----
+    If unspecified, all columns will be assumed to be known during prediction duration.
+    """
+
+    _tags = {
+        # packaging info
+        # --------------
+        "authors": ["yarnabrina", "fnhirwa"],
+        "maintainers": ["yarnabrina", "fnhirwa"],
+        "python_version": ">=3.9",
+        "python_dependencies": ["u8darts>=0.29"],
+        "python_dependencies_alias": {"u8darts": "darts"},
+        # estimator type
+        # --------------
+        "y_inner_mtype": "pd.DataFrame",
+        "X_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "enforce_index_type": pd.DatetimeIndex,
+        "handles-missing-data": False,
+        "capability:insample": False,
+    }
+
+    def __init__(
+        self: "_DartsRegressionModelsAdapter",
+        past_covariates: Optional[list[str]] = None,
+        num_samples: Optional[int] = 1000,
+    ) -> None:
+        if past_covariates is not None and not isinstance(past_covariates, list):
+            raise TypeError(
+                f"Expected past_covariates to be a list, found {type(past_covariates)}."
+            )
+        self.past_covariates = past_covariates
+        if num_samples is not None and not isinstance(num_samples, int):
+            raise TypeError(
+                f"Expected num_samples to be an integer, found {type(num_samples)}."
+            )
+        self.num_samples = num_samples
+
+        super().__init__()
+
+        # initialize internal variables to avoid AttributeError
+        self._forecaster = None
+
+    @staticmethod
+    def convert_dataframe_to_timeseries(dataset: pd.DataFrame):
+        """Convert dataset for compatibility with ``darts``.
+
+        Parameters
+        ----------
+        dataset : pandas.DataFrame
+            source dataset to convert from
+
+        Returns
+        -------
+        darts.TimeSeries
+            converted target dataset
+        """
+        import darts
+
+        dataset_copy = _handle_input_index(dataset)
+        return darts.TimeSeries.from_dataframe(dataset_copy)
+
+    @property
+    @abc.abstractmethod
+    def _lags_past_covariates(self):
+        """Get the lags_past_covariates value."""
+        raise NotImplementedError("This model does not support past covariates.")
```

Comment:
This `raise` is probably unnecessary, especially if we focus this class to be for regression models only.

Ref. https://unit8co.github.io/darts/generated_api/darts.models.forecasting.regression_model.html#darts.models.forecasting.regression_model.RegressionModel